app/src/main/java/org/thoughtcrime/securesms/onboarding/LinkDeviceActivity.kt

```diff
@@ -1,230 +1,346 @@
+            placeholder = { Text(stringResource(R.string.recoveryPasswordEnter)) },
+            colors = outlinedTextFieldColors(state.error != null),
+            singleLine = true,
+            keyboardActions = KeyboardActions(
+                onDone = { onContinue() },
+                onGo = { onContinue() },
+                onSearch = { onContinue() },
+                onSend = { onContinue() },
+            ),
+            isError = state.error != null,
+            shape = RoundedCornerShape(12.dp)
+        )
+        Spacer(Modifier.size(12.dp))
+        state.error?.let {
+            Text(
+                it,
+                modifier = Modifier.contentDescription(R.string.AccessibilityId_error_message),
+                style = MaterialTheme.typography.baseBold,
+                color = MaterialTheme.colors.error
+            )
         }
+        Spacer(Modifier.weight(2f))
+        OutlineButton(
+            text = stringResource(id = R.string.continue_2),
+            modifier = Modifier
+                .align(Alignment.CenterHorizontally)
+                .padding(horizontal = 64.dp, vertical = 20.dp)
+                .width(200.dp)
+        ) { onContinue() }
     }
 }
-// endregion
 
-// region Recovery Phrase Fragment
-class RecoveryPhraseFragment : Fragment() {
-    private lateinit var binding: FragmentRecoveryPhraseBinding
+fun Context.startLinkDeviceActivity() {
+    Intent(this, LinkDeviceActivity::class.java).let(::startActivity)
+}
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
-        binding = FragmentRecoveryPhraseBinding.inflate(inflater, container, false)
-        return binding.root
+@SuppressLint("UnsafeOptInUsageError")
+private fun buildAnalysisUseCase(
+    scanner: BarcodeScanner,
+    onBarcodeScanned: (String) -> Unit
+): ImageAnalysis = ImageAnalysis.Builder()
+    .setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
+    .build().apply {
+        setAnalyzer(Executors.newSingleThreadExecutor(), Analyzer(scanner, onBarcodeScanned))
     }
 
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-        with(binding) {
-            mnemonicEditText.imeOptions = EditorInfo.IME_ACTION_DONE or 16777216 // Always use incognito keyboard
-            mnemonicEditText.setRawInputType(InputType.TYPE_CLASS_TEXT)
-            mnemonicEditText.setOnEditorActionListener { v, actionID, _ ->
-                if (actionID == EditorInfo.IME_ACTION_DONE) {
-                    val imm = v.context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
-                    imm.hideSoftInputFromWindow(v.windowToken, 0)
-                    handleContinueButtonTapped()
-                    true
-                } else {
-                    false
+class Analyzer(
```

Comment:
Would it make more sense to put this in a separate file with the class name `QrCodeAnalyser` or such?

Reply:
yes, was left here in refactoring, but I think I moved it in chunk 2 :/ so I might prefer to leave here until then.

Reply:
Okay-doke - marking resolved.